xskillscore/tests/test_probabilistic.py

```diff
@@ -476,3 +478,225 @@ def test_rps_perfect_values(o, category_edges):
 def test_rps_dask(o_dask, f_prob_dask, category_edges):
     """Test that rps returns dask array if provided dask array"""
     assert rps(o_dask, f_prob_dask, category_edges=category_edges).chunks is not None
+
+
+@pytest.mark.parametrize(
+    "observation,forecast",
+    [
+        (
+            pytest.lazy_fixture("observation_1d_long"),
+            pytest.lazy_fixture("forecast_1d_long"),
+        ),
+        (
+            pytest.lazy_fixture("observation_3d"),
+            pytest.lazy_fixture("forecast_3d"),
+        ),
+    ],
+)
+@pytest.mark.parametrize("dim", ["time", None])
+@pytest.mark.parametrize("drop_intermediate", [True, False])
+@pytest.mark.parametrize("chunk", [True, False])
+@pytest.mark.parametrize("input", ["Dataset", "DataArray"])
+@pytest.mark.parametrize(
+    "return_results", ["all_as_tuple", "area", "all_as_metric_dim"]
+)
+def test_roc_returns(
+    observation,
+    forecast,
+    symmetric_edges,
+    dim,
+    return_results,
+    input,
+    chunk,
+    drop_intermediate,
+):
+    """testing keywords and inputs pass"""
+    if "Dataset" in input:
+        name = "var"
+        forecast = forecast.to_dataset(name=name)
+        observation = observation.to_dataset(name=name)
+    if chunk:
+        forecast = forecast.chunk()
+        observation = observation.chunk()
+
+    roc(
+        observation,
+        forecast,
+        symmetric_edges,
+        dim=dim,
+        drop_intermediate=drop_intermediate,
+        return_results=return_results,
+    )
+
+
+def test_roc_auc_score_random_forecast(
+    forecast_1d_long, observation_1d_long, symmetric_edges
+):
+    """Test that ROC AUC around 0.5 for random forecast."""
+    area = roc(
+        observation_1d_long,
+        forecast_1d_long,
+        symmetric_edges,
+        dim="time",
+        return_results="area",
+    )
+    assert area < 0.6
+    assert area > 0.4
+
+
+def test_roc_auc_score_perfect_forecast(forecast_1d_long, symmetric_edges):
+    """Test that ROC AUC equals 1 for perfect forecast."""
+    area = roc(
+        forecast_1d_long,
+        forecast_1d_long,
+        symmetric_edges,
+        drop_intermediate=False,
+        dim="time",
+        return_results="area",
+    )
+    assert area == 1.0
+
+
+@pytest.mark.parametrize("drop_intermediate", [False, True])
+def test_roc_auc_score_out_of_range_forecast(
+    forecast_1d_long, observation_1d_long, symmetric_edges, drop_intermediate
+):
+    """Test that ROC AUC equals 0.0 or 0.5 for out of range forecast."""
+    area = roc(
+        observation_1d_long,
+        xr.ones_like(forecast_1d_long) + 100,
+        symmetric_edges,
+        drop_intermediate=drop_intermediate,
+        dim="time",
+        return_results="area",
+    )
+    assert float(area) in [0.5, 0.0]
+
+
+@pytest.mark.parametrize("drop_intermediate", [False, True])
```

Comment:
Can you name that `drop_intermediate_bool`? We do that is other tests like `weight_bool`. Easier to follow